Source/ApiTemplate/Source/ApiTemplate/Program.cs

```diff
@@ -28,7 +28,7 @@ public static async Task<int> LogAndRunAsync(IHost host)
             }
 
             var hostEnvironment = host.Services.GetRequiredService<IHostEnvironment>();
-            hostEnvironment.ApplicationName = Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyProductAttribute>().Product;
+            hostEnvironment.ApplicationName = Assembly.GetExecutingAssembly()?.GetCustomAttribute<AssemblyProductAttribute>()?.Product;
```

Comment:
What do you think if we introduce a static class to wrap all these calls to get custom attributes everywhere? We could call it `ApplicationInfo` or `AssemblyInfo`.

We should also probably be using `typeof(Startup).Assembly` instead of `Assembly.GetExecutingAssembly()`.